bindings/rascal/utils/io.py

```diff
@@ -0,0 +1,387 @@
+    module = importlib.import_module(module_name)
+    class_ = getattr(module, class_name)
+    return class_
+
+
+def obj2dict_beta(cls, state):
+    """Take a python object cls with its state and return a dictionary that
+    can be used to create a copy of this object.
+
+    Parameters
+    ----------
+    cls : object
+
+    state : dictionary
+        Contains the state of cls, i.e. the parameters used to initialize cls
+        in a 'init_params' field and the rest of the data needed to recover
+        the current state in a 'data' field.
+
+    Returns
+    -------
+    dictionary
+        fully serialized version of cls to a dictionary
+    """
+    VERSION = BETA_VERSION
+    module_name = cls.__module__
+    class_name = cls.__name__
+    frozen = dict(version=VERSION, class_name=class_name,
+                  module_name=module_name,
+                  init_params=state['init_params'],
+                  data=state['data'])
+    return frozen
+
+
+def dict2obj_beta(data):
+    """Take data, a dictionary created by the obj2dict function, and creates
+    a python object as described.
+
+    Parameters
+    ----------
+    data : dictionary
+        [description]
+
+    Returns
+    -------
+    deserialized python object described by data
+    """
+    cls = get_class(data['module_name'], data['class_name'])
+    obj = cls(**data['init_params'])
+    obj._set_data(data['data'])
+    return obj
+
+
+def is_valid_object_dict_beta(data):
+    """check compatibility of data to be used in dict2obj_beta"""
+    valid_keys = ['version', 'class_name', 'module_name', 'init_params', 'data']
+    aa = []
+    if isinstance(data, dict):
+        for k in data:
+            if k in valid_keys:
+                aa.append(True)
+        if len(aa) == len(valid_keys):
+            return True
+        else:
+            return False
+    else:
+        return False
+
+
+obj2dict = {BETA_VERSION: obj2dict_beta}
+dict2obj = {BETA_VERSION: dict2obj_beta}
+is_valid_object_dict = {BETA_VERSION: is_valid_object_dict_beta}
+
+
+def get_current_io_version():
+    return CURRENT_VERSION
+
+
+def get_supported_io_versions():
+    return list(dict2obj.keys())
+
+
+class BaseIO(ABC):
+    """Interface of a Python class serializable by to_dict()
+
+    It corresponds to 3 methods:
+
+    + _get_init_params is expected to return a dictionary containing all the
+    parameters used by the __init__() methods.
+
+    + _get_data is expected to return a dictionary containing all the data
+    that is not set by the initialization of the class.
+
+    + _set_data is expected to set the data that has been extracted by _get_data
+    """
+    @abstractmethod
+    def _get_data(self):
+        return dict()
+
+    @abstractmethod
+    def _set_data(self, data):
+        pass
+
+    @abstractmethod
+    def _get_init_params(self):
+        return dict()
+
+
+def _get_state(obj):
+    if isinstance(obj, BaseIO) is True:
```

Comment:
```suggestion
    if isinstance(obj, BaseIO):
```
(`is True` is redundant)

Reply:
but more explicit

Reply:
No. It's just confusing.

Reply:
ok it is confusing to you and it is the opposite to me.

Reply:
does it really matter ?

Reply:
Yes. It's 100% redundant and _will_ be confusing to anyone else who reads the code, because it implies that `isinstance` (or the function defined later for testing a valid dict) might return something that's not a true/false value.

Reply:
also note that python implicitly evaluates many things to True in the if statements which can lead to unexpected behaviour.

Reply:
> will be confusing to anyone else who reads the code, because it implies that isinstance (or the function defined later for testing a valid dict) might return something that's not a true/false value.

Aren't you over reading into things. Also I would not presume so much about what someone else might think while reading this statement.

Reply:
PEP8 is explicitly against comparing Boolean values to True or False: https://www.python.org/dev/peps/pep-0008/#programming-recommendations. There is no more to say about this.